### PR TITLE
Add dependency on cozy-vcard

### PR DESCRIPTION
**Pull request by jacquarg**
- Rely on cozy-vcard for parsing, serializing and some methods of Contact.
- Change address fields format, with migration of old data in database.

### CODE REVIEW

**Comment by jsilvestre** [comment]
When I start the application, I get
```
TypeError: Object function VCardParser(vcf) {
      this.reset();
      if (vcf) {
        this.read(vcf);
      }
    } has no method 'fnToN'
  at ModelConstructor.Contact.getComputedN (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/server/models/contact.coffee:70:24)
  at ModelConstructor.Contact.afterInitialize (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/server/models/contact.coffee:36:15)
  at next (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb/lib/hooks.js:59:23)
  at ModelConstructor.trigger (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb/lib/hooks.js:54:9)
  at ModelConstructor.AbstractClass._initProperties (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb/lib/model.js:124:10)
  at ModelConstructor.AbstractClass (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb/lib/model.js:32:10)
  at new ModelConstructor (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb/lib/schema.js:218:23)
  at /Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb-cozy-adapter/lib/cozy_data_system.js:503:28
  at parseBody (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb-cozy-adapter/node_modules/request-json/main.js:26:12)
  at Request._callback (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb-cozy-adapter/node_modules/request-json/main.js:81:18)
  at Request.self.callback (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb-cozy-adapter/node_modules/request-json/node_modules/request/index.js:148:22)
  at Request.emit (events.js:98:17)
  at Request.<anonymous> (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb-cozy-adapter/node_modules/request-json/node_modules/request/index.js:876:14)
  at Request.emit (events.js:117:20)
  at IncomingMessage.<anonymous> (/Users/joseph/Documents/Developement/cozycloud/dev-env/contacts/node_modules/americano-cozy/node_modules/jugglingdb-cozy-adapter/node_modules/request-json/node_modules/request/index.js:827:12)
  at IncomingMessage.emit (events.js:117:20)
  at _stream_readable.js:944:16
  at process._tickCallback (node.js:442:13)
```
I did:
```
$ npm install
$ coffee server.coffee
```

**Comment by jacquarg** [comment]
I can't reproduce. I just : 
```
$ git clone https://github.com/jacquarg/cozy-contacts.git
$ ... fetch  branch
$ git branch cz-vcard
$ npm install # and build clients
$ coffee server.coffee
```

Which is the version of cozy-vcard in your node_module directory ? (should be  0.2.0)

**Comment by jsilvestre** [comment]
Ok, for some reason, I add to remove my node_modules and `npm install` it again oO Sorry!

**Comment by jsilvestre** [comment]
So the migration seems to work, all my old contacts now have their full name in the "first name" field.

I noticed a bug, when I change a field, the form re-renders, resulting in a data loss. How to reproduce:
* click the first name field
* add any character
* quickly click on the last name field, and quickly add other characters
* result: the characters in the last name field has disappeared

File: client/app/models/contact.coffee [hunk 1]
```diff
@@ -170,183 +162,10 @@ module.exports = class Contact extends Backbone.Model
...
-        when 'relation'
-            value = parts[1]
-            type = ANDROID_RELATION_TYPES[+parts[2]]
-            type = parts[3] if type is 'custom'
-            contact.addDP 'other', type, value
-
 Contact.fromVCF = (vcf) ->
+    ContactCollection = require 'collections/contact'
+    parser = new VCardParser()
+    parser.read(vcf)
```
Comment (jsilvestre) [inline client/app/models/contact.coffee]: useless parenthesis

**Comment by jacquarg** [comment]
I can't reproduce your bug, but still remove what should be the cause.
Needs build before merge !

**Comment by jsilvestre** [comment]
It's fixed indeed!
Another thing: when you select the full name display format, if you chose "last name first name", it actually displays "last name, first name". Can you remove the comma please? I know some country might do that, but the idea is to make it simpler and more robust.

**Comment by jsilvestre** [comment]
Thank you for the fixes. It happens an error is not caught when the image format is not properly encoded:
> Uncaught InvalidCharacterError: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.app.js:1838 module.exports.Contact.savePictureapp.js:4094 module.exports.ImporterView.addcontacts.contact.save.successvendor.js:10997 _.extend.save.options.successvendor.js:3165 jQuery.Callbacks.firevendor.js:3277 jQuery.Callbacks.self.fireWithvendor.js:8343 donevendor.js:8690 jQuery.ajaxTransport.send.callback

With this file: https://joseph-silvestre38.cozycloud.cc/public/files/files/28c0d6911e9b9e4f78dc89e5be06e3c0/attach/cozy-contacts-115-0-4.vcf

**Comment by jsilvestre** [comment]
Ok, it looks good now. There is one more step to do, it's testing the new contacts within Emails and Files/Photos (sharing modal). Give me some times to test that out.

**Comment by jsilvestre** [comment]
* [x] tests against Emails
* [x] tests against Files
* [x] tests against Photos
* [ ] sync tests (see https://github.com/cozy/cozy-sync/pull/61)

**Comment by jsilvestre** [comment]
@frankrousseau I tested the modification with everything that uses contacts. It's been working great so far. We need to merge Contacts and Sync simultaneously though.

@jacquarg can you add the build to the PR please?

**Comment by frankrousseau** [comment]
Ok once the build is added, we merge both!

**Comment by frankrousseau** [comment]
Good job @jacquarg! That was a lot of work!

**Pull request merged**